### DISCUSSION TITLE
CUDA Setup problems suddenly with Dreambooth

### DISCUSSION THREAD

**Discussion by AndyHTu** [d8ahazard/sd_dreambooth_extension#1382] [Q&A] [answered]
Hi guys,
I've been running into Dreambooth problems recently when troubleshooting all the extensions, I found that it has been causing problems with CUDA not working. Is this something that important to Dreambooth?   It pretty much breaks A1111 when I uninstall it.
Anyone else having this problem?

![cuda problems with dreambooth](https://github.com/d8ahazard/sd_dreambooth_extension/assets/115049891/e4b87e76-8c24-48c8-8cc7-e15e8db89c5a)

**Comment by mtnmecca** [comment]
issue was closed but a few days ago but after this latest update 11/18/23 the issue returned.  Reference  [Bug]: CUDA_SETUP: WARNING! libcudart.so not found in any environmental path. #1360

**Comment by UnconnectedBedna** [comment]
I got this when trying to install for the first time yesterday.

Everything else works perfectly in stable diffusion otherwise so I just assumed cuda was installed already.

Tried installing it from my repository but that version is newer than the files needed here.

So I ended up going for solution 2 in the bug report, install CUDA.
Problem 1: The file referenced is not named `cuda_install.sh` but there is a file called `install_cuda.sh` on the same repository.

Problem 2: Trying to use wget just scrambeled the file, I ended up using ctrl+c > ctrl+v from my web browser into a file I created myself. No idea what that was about, might also be completely unrelated to stable diffusion/dreambooth.

Problem 3: I found it a bit unclear what version, there is no mention in the wiki (witch is very sub par imo) but took a chance it is what PyTorch found, 118 so I installed that to a directory I made inside my cloned stable diffusion git repository, called it cuda.

Problem 4: The script tries to add stuff to my .bashrc, but since I am in a pyenv, that would not work and not something I would want anyway.

I then added following to my `web-user.sh`
```
export COMMANDLINE_ARGS="--xformers"
export REQS_FILE="./extensions/sd_dreambooth_extension/requirements.txt"
export XFORMERS_PACKAGE="xformers==0.0.21"
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:./cuda/cuda-11.8/targets/x86_64-linux/lib/
```

Again, do not know if this is the way to do it, the wiki is very confusing and lacking.

Stable diffusion starts, but I have not tried to do any training yet, if that fails, I might make a bug report, depends on how much time I am willing to give this.
The setup I now use, if dreambooth even works, idk yet, will most likely break sooner or later, and I am back to square 1.

But for now I need some sleep and hopefully someone who knows could return some feedback on what I wrote.

**Reply by AndyHTu** [reply]
Hi @UnconnectedBedna 
This is because Automatic1111 gets updated but the Dreambooth doesn't so you have to rollback to the right version to fix it.  I have a fix for that, but the better solution is to use the "Kohya_ss GUI" to do all your training. Its the best way to do model training right now because it is dedicated to just that.

**Reply by UnconnectedBedna** [reply, marked as answer]
Thank you so much for the response!

I am very new to this and is chugging away as it goes. So I really appreciate your response. There is sooo much to learn and it can be a bit overwhelming, like anything new and complicated you throw yourself into.

I took a look at the dates on this gituhub and yes, it seems you are correct. Did not check that before. xD
I just assumed this was supposed to work since it was on the plugins page on the newly installed SD.

I will obv look deeper into Kohya_ss GUI, after a quick search it seems to be a standalone aplication you run in a pyenv separate.
I am also on an arch based linux installation, so there is that, but I could probably get it to work, the question is just should I invest the time with my setup?
When getting to f.ex the [cuda toolkit](https://developer.nvidia.com/cuda-downloads?target_os=Linux&target_arch=x86_64), linked to in the installation instructions, there is nothing for arch, but there are for most other distros, not sure this is the way to go? I am not even sure what cuda stuff this is, is it the same I installed in my previous post so I can use that method instead? Because that method worked and I could choose exactly what version of cuda to install. :smile: 
Any input on that? :heart: 

Since you seem to be such a nice person. Do you have any recommendations on how to remove dreambooth? Should I just uninstall the plugin from the web gui, remove my manually installed stuff and then be done with it? Or will that leave a TON of stuff in my SD installation.
Sorry if the question is a bit MB on this github.

**Edit**
I just checked, and there seems there is no "remove" button after you have installed something. I do have backups of the data as it were before the installation, but I would rather just uninstall if it is possible?
I found [this](https://github.com/AUTOMATIC1111/stable-diffusion-webui/discussions/8052#discussioncomment-7095489) that says I might have to remove my env directory completely and let the script rebuild it again. IS this really the best way?

**Reply by AndyHTu** [reply]
It's not a problem at all. I like talking about this stuff :)
Welcome to the club, its a very fun time to learn this stuff. You came at a time where there are a lot to learn all at once, so just step back and take it little by little.  You can still use Dreambooth extension, but you'll need to get the older version, but like I said, just use Kohya_ss for all times of training. You can do Dreambooth, LoRa, Textual inversion. You can also write captions with it as well. It's everything you need to do machine learning for Stable Diffusion.  Join some of the communities on Discord and learn from other people, its a great way to network and learn things.  

I'm not overly technical, but with Dreambooth, you can just manually remove the extension from your Extensions folder in Automatic1111. 

stable-diffusion-webui > extensions >  Dreambooth > Delete the folder. 

I use Github Desktop to install everything, so I just remove the repository from Github desktop, which cleans it out.  This usually fix problems for me, but if it for some reason doesn't just do a fresh install, but save your Stable Diffusion base models so that you don't have to reinstall the environments again.  Having that already will save you a ton of time. 

I'm on discord a lot. If you want to share knowledge I'm more than happy to help you out, but it seems like you might be more technical than I am and would probably figure it out yourself eventually.

I made a guide on my site that might be easier to follow if you're interested.  You can DM me and I'll share it  as I don't want to shill here. lol

**Reply by UnconnectedBedna** [reply]
Wow, I hit the jackpot with hooking you.
We all know there are different mentality among the FOSS community, some have long beards and expects you to know how to compile and install linux from git, and then there are helpful and friendly people like you. Thank you.

I checked your webpage, but can not find any discord related things, wanna share a link?

**Reply by AndyHTu** [reply]
I have a discord  channel for my project only, which you might not be interested in, but my discord username is @andywear_bruh 
On discord.  If you want to add me.

I spend a lot of time in the invoke discord channel these days as I want to make that my WebUI this year. I'm always  grateful to make new friends so feel free to message me anytime!
